sympy/functions/special/bsplines.py

```diff
@@ -202,3 +202,60 @@ def bspline_basis_set(d, knots, x):
     """
     n_splines = len(knots) - d - 1
     return [bspline_basis(d, knots, i, x) for i in range(n_splines)]
+
+def interpolating_spline(d, x, xv='x', yv='y'):
+    """Return B-splines at ``x`` of degree ``d`` for given ``xv`` and corresponding ``yv`` values.
+
+    This function returns a list of Piecewise polynomials that are the
+    B-splines of degree ``d`` for the given points. This function
+    calls ``bspline_basis_set(n, knots, x)`` for different values of ``n``.
+
+    Examples
+    ========
+
+    >>> from sympy interpolating_spline
+    >>> from sympy.abc import x
+    >>> d = 2
+    >>> splines=interpolating_spline(d, x, [1,2,3], [1,2,3])
+    >>> splines
+    [4*Piecewise((-x**2/4 + x - 3/4, (x >= 1) & (x <= 3)), (0, True)) 
+     + Piecewise((x**2/4 - 3*x/2 + 9/4, (x >= 1) & (x <= 3)), (0, True)) 
+     + 3*Piecewise((x**2/4 - x/2 + 1/4, (x >= 1) & (x <= 3)), (0, True))
+    ] 
+
+    See Also
+    ========
+
+    bsplines_basis_set
+    """
+    from sympy import symbols
+    from sympy.solvers.solveset import linsolve
+    from sympy.matrices.dense import Matrix
+    
+    
+    ## IF no xv and yv values are passed as a argument
+    if isinstance(xv, str):
```

Comment:
I think you saw this in interpolating_poly method. But splines are different: their number of x-values is not determined by the degree. So this should be removed. If xv is not a list or a tuple of length at least d+1, that should be a ValueError; similarly for yv.

Reply:
Also check that xv and yv are of the same length.

Reply:
We should probably also check that the xv values are distinct and raise an error if they are not. Otherwise there's going some unhandled exception later.  `len(set(xv)) == len(xv)`  should be enough for this.